src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/analytics/custom.py

```diff
@@ -0,0 +1,229 @@
+            # pylint: disable=line-too-long
+            raise CLIError('Please specify --credential-user-name --password and --new-password in non-interactive mode.')
+
+    if not new_credential_user_password:
+        try:
+            new_credential_user_password = prompt_pass('New Credential Password:')
+        except NoTTYException:
+            # pylint: disable=line-too-long
+            raise CLIError('Please specify --credential-user-name --password and --new-password in non-interactive mode.')
+
+    update_params = DataLakeAnalyticsCatalogCredentialUpdateParameters(credential_user_password,
+                                                                       new_credential_user_password,
+                                                                       uri,
+                                                                       credential_user_name)
+    client.update_credential(account_name, database_name, credential_name, update_params)
+
+# job customizations
+# pylint: disable=too-many-arguments
+def submit_adla_job(client,
+                    account_name,
+                    job_name,
+                    script,
+                    runtime_version=None,
+                    compile_mode=None,
+                    compile_only=False,
+                    degree_of_parallelism=1,
+                    priority=1000):
+    contents = None
+    # script can be either script contents or a path to a script
+    # we attempt to always open the script path first, if that fails
+    # then treat it as contents
+    try:
+        with open(script, 'r') as f:
+            contents = f.read()
+    #pylint: disable=broad-except
+    except Exception:
+        contents = script
+
+    if not contents:
+        # pylint: disable=line-too-long
+        raise CLIError('Could not read script content from the supplied --script param. It is either empty or an invalid file. value: {}'.format(script))
+
+    job_properties = USqlJobProperties(contents)
+    if runtime_version:
+        job_properties.runtime_version = runtime_version
+
+    if compile_mode:
+        job_properties.compile_mode = compile_mode
+
+    submit_params = JobInformation(job_name,
+                                   JobType.usql,
+                                   job_properties,
+                                   degree_of_parallelism,
+                                   priority)
+    if compile_only:
+        return client.build(account_name, submit_params)
+
+    job_id = _get_uuid_str()
+
+    return client.create(account_name, job_id, submit_params)
+
+# pylint: disable=superfluous-parens
+def wait_adla_job(client,
+                  account_name,
+                  job_id,
+                  wait_interval_sec=5,
+                  max_wait_time_sec=-1):
+    if wait_interval_sec < 1:
+        # pylint: disable=line-too-long
+        logger.warning('wait times must be positive when polling jobs, setting to the default value of 5 seconds')
```

Comment:
It would be more consistent with the rest of the CLI to simply error out on this condition.